From 39bf29657daa2907e2c96d0a92874e35868e9fe0 Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Sun, 23 Aug 2020 20:02:44 +0300
Subject: [PATCH] [SW-17918] gaudi: security: configure pcie drain

Configure PCIe drain registers.
This is part of registers initialization which was previosuly done
in host, and is now being transferred from host to FW.
This is also a preparation for security registers access limitations.

Change-Id: I776b9cd7ce4cedc873723be9666dcf8f7b4c7fb7
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index d678711..e59076f 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -127,3 +127,9 @@
 
 	return 0;
 }
+
+void hl_pcie_draini_init(void)
+{
+	regmap_writel(0, mmPCIE_WRAP_LBW_DRAIN_CFG);
+	regmap_writel(0, mmPCIE_WRAP_HBW_DRAIN_CFG);
+}
diff --git a/gaudi/common.h b/gaudi/common.h
index 3deb131..132277a 100644
--- a/gaudi/common.h
+++ b/gaudi/common.h
@@ -23,6 +23,8 @@
 #define mmSRAM_W_PLL_BASE		0x4C8000
 #define mmMESH_E_PLL_BASE		0x4E7000
 #define mmSRAM_E_PLL_BASE		0x4E8000
+#define mmPCIE_WRAP_HBW_DRAIN_CFG	0xC01D54
+#define mmPCIE_WRAP_LBW_DRAIN_CFG	0xC01D5C
 #define mmPSOC_CPU_PLL_BASE		0xC70000
 #define mmPSOC_MME_PLL_BASE		0xC71000
 #define mmPSOC_PCI_PLL_BASE		0xC72000
diff --git a/ppboot/ppboot.c b/ppboot/ppboot.c
index 8c8baff..78994fb 100644
--- a/ppboot/ppboot.c
+++ b/ppboot/ppboot.c
@@ -143,6 +143,8 @@
 
 __weak void hl_init_pcie_phy_fw(void) {}
 
+__weak void ppboot_pcie_drain_init(void) {}
+
 static inline void ppboot_version(void)
 {
 	preboot_puts("Ppboot " VERSION_STR "\n");
@@ -164,6 +166,7 @@
 	preboot_sync_btl();
 	hl_uart_init();
 	ppboot_pcie_init();
+	ppboot_pcie_drain_init();
 	ppboot_version();
 	rst_src_dump();
 
